servers/cromwell/jobs/controllers/jobs_controller.py

```diff
@@ -206,6 +206,9 @@ def query_jobs(body, **kwargs):
         offset = offset + page_size
         page = page_from_offset(offset, page_size)
 
+    if query_page_size >= total_results:
```

Comment:
From looking above, total_results is the total number of all workflows matching the given query (unpaginated). If so, you don't want that.

Is there a test for this code that you can iterate against?